pkg/deb/debian.module/rules-noarch.in

```diff
@@ -3,8 +3,18 @@
 # Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
-export DEB_BUILD_MAINT_OPTIONS=hardening=+all,-pie
+include /usr/share/dpkg/vendor.mk
+ifeq ($(shell $(call dpkg_vendor_derives_from,ubuntu)),yes)
+DISTROREL_MAJOR := $(shell lsb_release -rs | cut -d. -f 1)
+ifeq ($(shell test $(DISTROREL_MAJOR) -ge 24; echo $$?), 0)
+export DEB_CFLAGS_MAINT_APPEND=-fPIC
+else
+export DEB_CFLAGS_MAINT_APPEND=-Wp,-D_FORTIFY_SOURCE=2 -fPIC
+endif
+else
 export DEB_CFLAGS_MAINT_APPEND=-Wp,-D_FORTIFY_SOURCE=2 -fPIC
+endif
```

Comment:
Anyway we can loose one of those duplicate branches?, i.e only having one of

```
export DEB_CFLAGS_MAINT_APPEND=-Wp,-D_FORTIFY_SOURCE=2 -fPIC
```

Ditto for the others...

Or do you envisage them diverging at some point?

Reply:
Oh, it looks like we can drop most of the patch for Ubuntus, since they seem to be using FORTIFY_SOURCE=2 for all their releases since ages, and moved to FORTIFY_SOURCE=3 for 24.04:

https://salsa.debian.org/toolchain-team/gcc/-/commit/ae01f86e0731d2661946866464939aba79b49da9

Reply:
I've updated the patchset accordingly. Thanks!